spec/views/reports/regions/_header.html.erb_spec.rb

```diff
@@ -0,0 +1,66 @@
+require "rails_helper"
+
+RSpec.describe "reports/regions/_header.html.erb", type: :view do
+  let(:organization) { create(:organization) }
+  let(:current_admin) { create(:admin, :manager) }
+  let(:organization_region) { create(:region, name: "Organization", region_type: "organization", path: "organization") }
+  let(:organization_region_1) { create(:region, name: "facility", region_type: "facility", path: "organization.state.district") }
+  let(:current_period) { Period.current }
+
+  before do
+    inject_controller_helper_methods(Reports::RegionsController, view)
```

Comment:
**suggestion**: If we choose to keep this a view spec, we can include just the helpers we need, like this:

```ruby
  helper do
    def accessible_region?(*args)
      true
    end

    def active_action?(*args)
      false
    end
  end
```

**thought** (no action needed): I think going with the view spec for now is okay, but down the line, we should perhaps break down this view into more components so that each individual part is easier to test and we won't need to stub helpers like this.

Reply:
Also when I tried to do this I realised we might have to stub out the other components like so:

```ruby
    stub_const("Dashboard::Hypertension::DownloadsDropdownComponent", Class.new {
      def initialize(*); end
      def render_in(*); ""; end
    })
    stub_const("Dashboard::Diabetes::DownloadsDropdownComponent", Class.new {
      def initialize(*); end
      def render_in(*); ""; end
    })
```

But adding these stubs explicitly seems better than the magic involved in injecting controller methods via monkey patching.

Reply:
@tfidfwastaken I think this is not related to injecting part as components were accessible in views, but the helper methods weren't, even if we don't stub this components or inject helpers it will work eventually as components are accessible in views.